tools/rosout/rosout.cpp

```diff
@@ -144,21 +144,27 @@ class Rosout
     ss << msg->name << " ";
     ss << "[" << msg->file << ":" << msg->line << "(" << msg->function << ")] ";
 
-    ss << "[topics: ";
-    std::vector<std::string>::const_iterator it = msg->topics.begin();
-    std::vector<std::string>::const_iterator end = msg->topics.end();
-    for ( ; it != end; ++it )
+    // check parameter server for omit_topics_from_logs flag
+    int omit_topics_from_logs;
+    node_.param("/rosout/omit_topics_from_logs", omit_topics_from_logs, 0);
```

Comment:
This parameter should probably be checked once on startup, not every time a message arrives.

(I'm also not in love with the name, but I'm not sure what else to suggest there. Maybe others have ideas?)

Reply:
rosout is kinda for logging so maybe just "omit_topics" is enough for us to be clear what the parameter does.

Reply:
`log_topic_names=false` is another possibility, but @dirk-thomas can make the final call.

Reply:
changed for now to omit_topics

Reply:
checking the parameter within the init() function was not working reliably for me, so instead I used "ros::NodeHandle::getParamCached()" which locally caches the parameter

Reply:
Okay, hmm. I could see that, given that there would be a race between this and the parameter server coming up.

In that case, my concern is about what happens when the parameter is absent. Does `getParamCached` support a default value? Can it distinguish between "param server is missing" and "key not on param server", and does its caching behaviour do the right thing?

Reply:
It doesn't support a default value, but I initialize the Rosout member `omit_topics_` to false in the constructor so that if `getParamCached` cannot find the parameter then it behaves as normal and topics are still printed. It behaves in the same way if it cannot contact the parameter server (http://docs.ros.org/diamondback/api/roscpp/html/param_8cpp_source.html#l00162).

Reply:
Okay, thanks for the thorough looking. In that case, LGTM.

Reply:
@mikepurvis can I go ahead and merge?